gymnasium/spaces/discrete.py

```diff
@@ -84,13 +99,13 @@ def sample(self, mask: MaskNDArray | None = None) -> np.int64:
                 np.logical_or(mask == 0, valid_action_mask)
             ), f"All values of a mask should be 0 or 1, actual values: {mask}"
             if np.any(valid_action_mask):
-                return self.start + self.np_random.choice(
```

Comment:
You should be able to pass a type to `choice` or use `asdtype(self.dtype)`

Reply:
unfortunately that returns an integer and it doesn't have `asdtype` or `astype`.

Reply:
Let me know if you have any other idea, otherwise would have to go with `self.dtype.type()`

Reply:
no worries, `self.dtype.type` is fine